docs/decisions/0001-directory-structure.md

```diff
@@ -0,0 +1,70 @@
+# Directory Structure
+
+## Summary
+
+We want to formalize the file and directory structure for the `cwms-python` module source code.
+
+## Considered Options
+
+Each top-level CWMS Data API endpoint corresponds to a `cwms-python` submodule. Nested endpoints would then map to a sub-directory within the submodule. The table below lists some example API endpoints and the corresponding submodule.
+
+| API Endpoint                                  | CWMS Python Module           |
+|-----------------------------------------------|------------------------------|
+| `/cwms-data/location/category`                | `cwms.location.category`     |
+| `/cwms-data/location/group`                   | `cwms.location.group`        |
+| `/cwms-data/locations`                        | `cwms.locations`             |
+| `/cwms-data/levels`                           | `cwms.levels`                |
+| `/cwms-data/specified-levels`                 | `cwms.specified_levels`      |
+| `/cwms-data/timeseries`                       | `cwms.timeseries`            |
+| `/cwms-data/timeseries/category`              | `cwms.timeseries.category`   |
+| `/cwms-data/timeseries/identifier-descriptor` | `cwms.timeseries.identifier` |
+| `/cwms-data/timeseries/group`                 | `cwms.timeseries.group`      |
+
+### Special Cases
+
+The direct mapping outlined above works well for most of the API endpoints. However, there are a couple cases which may warrant special consideration.
+
+#### Location Related Endpoints
+
+The locations, location categories, and location groups API endpoints return related data, despite having different URL paths (`/cwms-data/locations` vs. `/cwms-data/location/{category,group})`). From a `cwms-python` perspective, it may make more sense (and be less confusing to the user) if the corresponding functions were defined in the same submodule. This could be either `cwms.location` or `cwms.locations`, but since we are using the plural case for other endpoints (`cwms.levels`, `cwms-timeseries`, etc), `cwms.locations` would be the more natural choice.
```

Comment:
Similar to my comment about the `level`, the locations are referred to as "physical locations" so we could rename the module to `cwms.locations.physical_locations` to avoid confusion.

REST API design is more opinionated about where the plural is put, but I agree that the python API does not need to be constrained as such.